src/config.h

```diff
@@ -84,7 +84,7 @@ struct Config {
     // TextDocumentClientCapabilities.documentSymbol.hierarchicalDocumentSymbolSupport
     bool hierarchicalDocumentSymbolSupport = false;
     // TextDocumentClientCapabilities.completion.completionItem.snippetSupport
-    bool snippetSupport = false;
+    bool snippetSupport = true;
```

Comment:
Did you find an editor which supports LSP-style snippets but does not announce it?

Reply:
No. This is for force **disable** the snippet support instead of force enable.

Before this change `snippetSupport` is override by `params.capabilities.completion.completionItem.snippetSupport` in `initialize`.
I believe it makes sense to enable snippet support only when `params.capabilities.completion.completionItem.snippetSupport == true && client.snippetSupport == true` as some clients does not support configuring the capabilities it announce.

Reply:
OK. I think you mean even if the client announces that it supports snippets, i.e. `params.capabilities.completion.completionItem.snippetSupport == true`

Users should be able to disable it with `client.snippetSupport: false`

I think this change also applies to `hierarchicalDocumentSymbolSupport`